lib/pages/infoPages/vendorPage.dart

```diff
@@ -255,101 +415,145 @@ class _VendorButtonRowState extends State<VendorButtonRow> {
   void initialize() async {
     user = await _auth.currentUser();
     _userRef = FirebaseDatabase().reference().child("Users").child(user.uid);
+    _vendorRef = FirebaseDatabase().reference().child("Vendors").child(widget.vendor.key);
     _userRef.child("following").onValue.listen((data){
       if (data.snapshot != null){
         setState(() {
-          _following = data.snapshot.value[widget.vendorID]?? false;
+          _following = data.snapshot.value[widget.vendor.key]?? false;
         });        
       }
     });
   }
 
   @override
   Widget build(BuildContext context) {
-    return Row(
-      crossAxisAlignment: CrossAxisAlignment.center,
-      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-      children: <Widget>[
-        ButtonTheme(
-          minWidth: MediaQuery.of(context).size.width*0.3,
-          height: 80,
-          child: FlatButton(
-            child: Column(
-              children: <Widget>[
-                Icon(Icons.map, color: Colors.white),
-                Container(height: 4.0),
-                Text("Directions", style: TextStyle(color: Colors.white)),
-              ],
-            ), 
-            shape:  RoundedRectangleBorder(borderRadius: new BorderRadius.circular(12.0)),
-            onPressed: () {
-              openMap();
-            },
-            color: SavourColorsMaterial.savourGreen,
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(8.0, 8.0, 8.0, 28.0),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: <Widget>[
+          ButtonTheme(
+            minWidth: MediaQuery.of(context).size.width*0.3,
+            height: 80,
+            child: FlatButton(
+              child: Column(
+                children: <Widget>[
+                  Icon(Icons.map, color: Colors.white),
+                  Container(height: 4.0),
+                  Text("Directions", style: TextStyle(color: Colors.white)),
+                ],
+              ), 
+              shape:  RoundedRectangleBorder(borderRadius: new BorderRadius.circular(12.0)),
+              onPressed: () {
+                openMap();
```

Comment:
should this be private?

Reply:
Are you asking if the openMap() function should be private?